### ISSUE TITLE
Bug: Stock overview displays nothing (crashes the JavaScript library for interactive tables)

### ISSUE THREAD

**Issue by muki** [grocy/grocy#1755]
_Please describe the bug as detailed as possible, provide the steps how to reproduce it and maybe attach screenshots where useful._

I started off with a fresh installation of Grocy 3.1.3 and then used [this app](https://github.com/patzly/grocy-android) (I checked quite hard, the issue doesn't seem to come from there, especially since the rest of Grocy works fine) to add inventory by scanning the barcodes and filling out all the necessary fields. The items are stored in the database correctly, I can confirm that. I can edit them individually, manipulate them on the shopping list (adding those below min amount works as well), but if I open the Stock overview tab the table crashes. The yellow, gray, red, and blue filters on top contain no text, the table contains no items and the console shows the following error thrown by the first line of stockoverview.js: `Uncaught TypeError: can't access property "_DT_CellIndex", e is undefined`.

The issue began with the first entered item. If you think it would help with diagnosing the bug I can manipulate my data or provide a state and/or example stock overview HTML output.

_Please also check if your bug was maybe already fixed by searching closed issues here or by trying to reproduce your problem on the [pre-release demo](https://demo-prerelease.grocy.info/) (use a *private demo instance* if you want to make your example persistent)._

It's been a couple of weeks now and I tried various things I could find in the issues here relating to `_DT_CellIndex` but nothing seemed to work. Also restarted several times as well as manually removed and re-added the barcode to all the items (through the admin interface for managing master data)

_If your problem is not reproducible, it's most likely not a bug - please use the [r/grocy subreddit](https://www.reddit.com/r/grocy) for general questions / help._

I tried reproducing the bug with the private instance by hand, but am unable to do so. I can further diagnose this if you tell me where to look. It's very strange that everything seems to be working except the stock overview. It loads the HTML table, but the JS dies.

**Comment by berrnd** [comment]
> the console shows the following error thrown by the first line of stockoverview.js: `Uncaught TypeError: can't access property "_DT_CellIndex", e is undefined`.

The error message is pretty clear: A not existent table column is referenced by the JavaScript - I don't know how that's possible.

We had that once before in #1202, where it was about disabled features (`config.php` `FEATURE_FLAG` options).

Maybe it's a specific situation related to that, then feel free to provide the reproducible example. Unless it is reproducible, it makes no sense to track it as a bug (as mentioned on the issue template).

Can also be browser caching or related things to that if you ever used a different version before...

**Comment by berrnd** [comment]
And if all doesn't help, you cannot create a reproducible example on the demo instance and you really think this is a grocy and not a user or configuration problem: Feel free to send me an example database (upload it here or send me a mail) + `config.php` (all except path related options) and I'm happy to look into it.

**Comment by muki** [comment]
Thank you for taking the time to review this.

I tried toggling the FEATURE_FLAG from #1202 as well as all the other that seemed somehow "related" but nothing worked. I've been at it for a few weeks now, all the while the rest of Grocy seems to be working fine. It's very strange.

I copied the sqlite database to both of the docker versions (from the readme) and it crashed completely (returning a 500 error). I do not know enough PHP to be able to figure out what exactly the error was (or where it was logged :see_no_evil:). In any case, I did not have too much data in my instance so I threw it all away and installed a new version where I started off with manual entry and so far things seem to be working. I still have the old "broken" database file and can send it to you, but as you mentioned since I'm unable to reproduce this the ticket might as well be closed.

I'll hopefully find some time in February to try and reproduce the issue on a fresh installation with reliable and clear steps. Until then, take care, and thank you once again!
